### ISSUE TITLE
bug with argparse in buildtest build

### ISSUE THREAD

**Issue by shahzebsiddiqui** [buildtesters/buildtest#212]
There is some bug when running ``buildtest build --help`` there is a TypeError issue which i dont get what happened. I haven't had time to look into this as we need to change the way argparse is setup. For now putting this in to fix.

```
(docs) ssi29@ag-mxg-hulk090> buildtest build -h
Traceback (most recent call last):
  File "/u/users/ssi29/.local/share/virtualenvs/docs-dnVBLcEp/bin/buildtest", line 11, in <module>
    load_entry_point('buildtest-framework', 'console_scripts', 'buildtest')()
  File "/mxg-hpc/users/ssi29/buildtest-framework/buildtest/main.py", line 20, in main
    parser.parse_options()
  File "/mxg-hpc/users/ssi29/buildtest-framework/buildtest/menu/__init__.py", line 76, in parse_options
    args = self.parser.parse_args()
  File "/mxg-hpc/users/ssi29/easybuild/software/Python/3.7.4-GCCcore-8.3.0/lib/python3.7/argparse.py", line 1749, in parse_args
    args, argv = self.parse_known_args(args, namespace)
  File "/mxg-hpc/users/ssi29/easybuild/software/Python/3.7.4-GCCcore-8.3.0/lib/python3.7/argparse.py", line 1781, in parse_known_args
    namespace, args = self._parse_known_args(args, namespace)
  File "/mxg-hpc/users/ssi29/easybuild/software/Python/3.7.4-GCCcore-8.3.0/lib/python3.7/argparse.py", line 1969, in _parse_known_args
    positionals_end_index = consume_positionals(start_index)
  File "/mxg-hpc/users/ssi29/easybuild/software/Python/3.7.4-GCCcore-8.3.0/lib/python3.7/argparse.py", line 1946, in consume_positionals
    take_action(action, args)
  File "/mxg-hpc/users/ssi29/easybuild/software/Python/3.7.4-GCCcore-8.3.0/lib/python3.7/argparse.py", line 1855, in take_action
    action(self, namespace, argument_values, option_string)
  File "/mxg-hpc/users/ssi29/easybuild/software/Python/3.7.4-GCCcore-8.3.0/lib/python3.7/argparse.py", line 1152, in __call__
    subnamespace, arg_strings = parser.parse_known_args(arg_strings, None)
  File "/mxg-hpc/users/ssi29/easybuild/software/Python/3.7.4-GCCcore-8.3.0/lib/python3.7/argparse.py", line 1781, in parse_known_args
    namespace, args = self._parse_known_args(args, namespace)
  File "/mxg-hpc/users/ssi29/easybuild/software/Python/3.7.4-GCCcore-8.3.0/lib/python3.7/argparse.py", line 1987, in _parse_known_args
    start_index = consume_optional(start_index)
  File "/mxg-hpc/users/ssi29/easybuild/software/Python/3.7.4-GCCcore-8.3.0/lib/python3.7/argparse.py", line 1927, in consume_optional
    take_action(action, args, option_string)
  File "/mxg-hpc/users/ssi29/easybuild/software/Python/3.7.4-GCCcore-8.3.0/lib/python3.7/argparse.py", line 1855, in take_action
    action(self, namespace, argument_values, option_string)
  File "/mxg-hpc/users/ssi29/easybuild/software/Python/3.7.4-GCCcore-8.3.0/lib/python3.7/argparse.py", line 1037, in __call__
    parser.print_help()
  File "/mxg-hpc/users/ssi29/easybuild/software/Python/3.7.4-GCCcore-8.3.0/lib/python3.7/argparse.py", line 2474, in print_help
    self._print_message(self.format_help(), file)
  File "/mxg-hpc/users/ssi29/easybuild/software/Python/3.7.4-GCCcore-8.3.0/lib/python3.7/argparse.py", line 2458, in format_help
    return formatter.format_help()
  File "/mxg-hpc/users/ssi29/easybuild/software/Python/3.7.4-GCCcore-8.3.0/lib/python3.7/argparse.py", line 284, in format_help
    help = self._root_section.format_help()
  File "/mxg-hpc/users/ssi29/easybuild/software/Python/3.7.4-GCCcore-8.3.0/lib/python3.7/argparse.py", line 215, in format_help
    item_help = join([func(*args) for func, args in self.items])
  File "/mxg-hpc/users/ssi29/easybuild/software/Python/3.7.4-GCCcore-8.3.0/lib/python3.7/argparse.py", line 215, in <listcomp>
    item_help = join([func(*args) for func, args in self.items])
  File "/mxg-hpc/users/ssi29/easybuild/software/Python/3.7.4-GCCcore-8.3.0/lib/python3.7/argparse.py", line 215, in format_help
    item_help = join([func(*args) for func, args in self.items])
  File "/mxg-hpc/users/ssi29/easybuild/software/Python/3.7.4-GCCcore-8.3.0/lib/python3.7/argparse.py", line 215, in <listcomp>
    item_help = join([func(*args) for func, args in self.items])
  File "/mxg-hpc/users/ssi29/easybuild/software/Python/3.7.4-GCCcore-8.3.0/lib/python3.7/argparse.py", line 525, in _format_action
    help_text = self._expand_help(action)
  File "/mxg-hpc/users/ssi29/easybuild/software/Python/3.7.4-GCCcore-8.3.0/lib/python3.7/argparse.py", line 614, in _expand_help
    return self._get_help_string(action) % params
TypeError: unsupported operand type(s) for %: 'tuple' and 'dict'

```

**Comment by vsoch** [comment]
Ah, I thought that you had fixed this! If you've already given it a try I can definitely take a look.

**Comment by shahzebsiddiqui** [comment]
Nope i didn't fix this, i realized this a few PRs back since we don't have any regtest for these it was undetected but I knew about this and didn't bother fixing it yet. It will be important to get this working. I personally don't like the way i have setup argparse

**Comment by vsoch** [comment]
okay, it sounds like it's been frustrating - let me put this in my queue of things to do, maybe fresh eyes will help.

And I think you did okay with argparse, it's clear how it's setup! This is likely just a bug.

**Comment by vsoch** [comment]
I've fixed this locally in my PR.

**Comment by vsoch** [comment]
Also, when I open a PR for this I'll do rebase (locally) and merge the commit cleanly so we don't need to do a squash and then I would lose credit for the work (like what happened in the schemas repo).

**Comment by vsoch** [comment]
This is fixed with #219
